yarn-project/noir-contracts/src/contracts/easy_private_voting_contract/src/main.nr

```diff
@@ -0,0 +1,117 @@
+    use dep::aztec::{
+	context::{PrivateContext, Context},
+    oracle::get_secret_key::get_secret_key, // used to compute nullifier
+    selector::compute_selector, // used to compute function selector for calling a function 
+	 state_vars::{ map::Map, public_state::PublicState,}, 
+     types::type_serialization::{ // serialization methods for using booleans and aztec addresses
+	    bool_serialization::{BoolSerializationMethods, BOOL_SERIALIZED_LEN},
+	    aztec_address_serialization::{AztecAddressSerializationMethods, AZTEC_ADDRESS_SERIALIZED_LEN},
+        field_serialization::{ FieldSerializationMethods, FIELD_SERIALIZED_LEN},
+	}, 
+    types::address::{AztecAddress},
+    constants_gen::EMPTY_NULLIFIED_COMMITMENT,
+    };
+    // docs:end:imports
+    // docs:start:storage_struct
+    struct Storage {
+    admin: PublicState<AztecAddress, AZTEC_ADDRESS_SERIALIZED_LEN>, // admin can end vote
+    tally: Map<PublicState<Field, FIELD_SERIALIZED_LEN>>, // we will store vote as key and number of votes as value
+    voteEnded: PublicState<bool, BOOL_SERIALIZED_LEN>, // voteEnded is boolean
+}
+    // docs:end:storage_struct
+    // docs:start:storage_impl
+    impl Storage {
+	fn init(context: Context) -> Self {
+	    Storage {
+		 admin: PublicState::new(
+		    context, 
+            1, // storage slot. this can be anything except 0. it is hashed, and hash on 0 = 0
+            AztecAddressSerializationMethods,
+		), 
+        tally: Map::new(
+		    context, 
+            2, 
+            |context, slot| {
+			PublicState::new(
+			    context,
+                slot, 
+                FieldSerializationMethods,
+			)
+		    },
+		 ),
+		  voteEnded: PublicState::new(
+		    context, 
+            3, 
+            BoolSerializationMethods,
+		)
+	    } }
+	}
+    // docs:end:storage_impl
+    // docs:start:constructor
+    #[aztec(private)] // annotation to mark function as private and expose private context
+    fn constructor(admin: AztecAddress) { // called when contract is deployed
+        context.call_public_function(
+            // we cannot update public state directly from private function but we can call public function (which queues it)
+            context.this_address(),// address that calls public function. here we are using contract address for privacy
+            compute_selector("_initialize((Field))"), // selector of  set_admin()
+            [admin.address] // parameters for set_admin()
+        );
+    }
+    // docs:end:constructor
+    // docs:start:initialize
+    #[aztec(public)] // annotation to mark function as public and expose public context
+    internal fn _initialize(admin: AztecAddress) { // internal - can only be called by contract
+        storage.admin.write(admin);
+        storage.voteEnded.write(false);
+    }
+    // docs:end:initialize
+    // docs:start:cast_vote
+    #[aztec(private)] // annotation to mark function as public and expose private context
+    fn cast_vote(voteId: Field) {
```

Comment:
I know I suggested it, but `voteID` doesn't sound right

Instead we can call it a `candidate`/`nominee`/`contender`?

Reply:
I actually quite like voteId - it doesn't have to be for a person, it could be yes/no, or some other proposal. but can change if you definitely think it sounds off

Reply:
yea can you please change 🙏 ? 
`voteID` makes me think there are multiple proposals. For each proposal i can either vote or not vote. But what am I exactly voting? For a given voteID am I voting for it or against????

Instead your contract is for a campaign and people vote for one fo the options. Hence "candidate"

Reply:
youre right. updated